server/routes.go

```diff
@@ -199,11 +199,10 @@ func (s *Server) GenerateHandler(c *gin.Context) {
 				c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
 				return
 			}
-
-			b.WriteString(s)
+			prompt = s + b.String()
```

Comment:
instead of string concat, it would be better to move this `if req.Context != nil` block before the `tmpl.Execute` block

```go
var b bytes
if req.Context != nil {
    s, err := r.Detokenize(c.Request.Context(), req.Context)
    ...
    b.WriteString(s)
}

if err := tmpl.Execute(&b, values); err != nil {
    ...
}
```